packages/ckit/src/providers/mercury/MercuryProvider.ts

```diff
@@ -227,6 +227,46 @@ export class MercuryProvider extends AbstractProvider {
     return acc.cells;
   }
 
+  async collectUdtCellsByMinCkb(
```

Comment:
maybe it is an unnecessary collector here, we can refactor `collectCells` to implement it and avoid repeating self

```ts
class Provider {
  // collectCells({ searchKey }: { searchKey: SearchKey }, takeWhile_: (cell: Cell[]) => boolean): Promise<Cell[]>

  collectCells({ searchKey }: { searchKey: SearchKey }, takeWhile_: (cell: Cell[]) => boolean, options?: { inclusive?: boolean }): Promise<Cell[]>

}
```

```ts
// usage
collectCells(
  searchKey, 
  (acc) => sumCapacity(acc).lt(minimalCapacity), 
  { inclusive: true }
)
```